wallet/src/types.rs

```diff
@@ -52,6 +57,8 @@ impl Default for WalletConfig {
 			chain_type: Some(ChainTypes::Testnet3),
 			api_listen_interface: "127.0.0.1".to_string(),
 			api_listen_port: 13415,
+			owner_api_basic_auth: Some(true),
+			owner_api_secret: thread_rng().sample_iter(&Alphanumeric).take(20).collect(),
```

Comment:
Do I understand correctly we regenerate secret every time but it will be used only if it was explicitly persisted?

Reply:
No this is generated for every new config file. On restart it will persist.